pkg/controller/common.go

```diff
@@ -1887,14 +1973,16 @@ func (vq *VolumeReplicaRequestsPriorityQueue) DrainQueue() []*ReplicaRequest {
 
 ///Removes replica requests from the priority queue and adds to operation queue.
 func (c *SharedState) tryCreateFailedReplicas(ctx context.Context, requestor operationRequester) error {
+	workflow, _ := azureutils.GetWorkflowFromContext(ctx)
```

Comment:
This seems like a standalone workflow triggered from a specific event. I believe it make sense to create a new workflow here rather than rely on the caller to provide it.

Reply:
See my comment in the node availability controller.